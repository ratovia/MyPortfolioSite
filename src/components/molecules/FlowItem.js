import React from "react"
import styled from "styled-components"
import { Batch } from "../atoms/Batch"
import { Flow } from "../atoms/Flow"
import { TypographyH5, TypographyH6 } from "../atoms/Typography"

const FlowArea = styled.div`
  width: 100%;
  display: flex;
  position: relative;
`

const DateArea = styled.div`
  margin-top: 100px;
  min-width: 100px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`

const LineArea = styled.div`
  padding: 0 50px;
`

const TextArea = styled.div`
  margin-top: 100px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-bottom: 100px;
`

const TextItem = styled.div`
  max-width: 700px;
  margin-bottom: 15px;
`
const BatchArea = styled.span`
  margin-right: 10px;
`
export const FlowItem = ({
  startDate,
  endDate,
  orgName,
  period,
  content,
  batchs,
}) => {
  return (
    <FlowArea>
      <DateArea>
        <TypographyH6 label={startDate} color="#C7C7C7"></TypographyH6>
        <TypographyH6 label="ã€œ" color="#C7C7C7"></TypographyH6>
        <TypographyH6 label={endDate} color="#C7C7C7"></TypographyH6>
      </DateArea>
      <LineArea>
        <Flow></Flow>
      </LineArea>
      <TextArea>
        <TextItem>
          <TypographyH5 label={orgName} color="white"></TypographyH5>
          <TypographyH6 label={period} color="#C7C7C7"></TypographyH6>
        </TextItem>
        <TextItem>
          {batchs.map(batch => {
            return (
              <BatchArea key={batch}>
                <Batch label={batch}></Batch>
              </BatchArea>
            )
          })}
        </TextItem>
        <TextItem>
          <TypographyH6 label={content} color="white"></TypographyH6>
        </TextItem>
      </TextArea>
    </FlowArea>
  )
}
