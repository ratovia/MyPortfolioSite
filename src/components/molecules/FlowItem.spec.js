import React from "react"
import renderer from "react-test-renderer"
import { FlowItem } from "./FlowItem"
describe("FlowItem", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <FlowItem
          startDate={"2019年8月"}
          endDate={"現在"}
          orgName={"株式会社xxx"}
          period={"1年間"}
          content={
            "helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld"
          }
          batchs={["営業", "エンジニア", "セールス", "労務"]}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
