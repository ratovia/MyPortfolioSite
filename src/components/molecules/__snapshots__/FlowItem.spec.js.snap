// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowItem renders correctly 1`] = `
<div
  className="sc-furvIG dEzHig"
>
  <div
    className="sc-pVTma ismugv"
  >
    <h6
      className="sc-hKwCoD hglaFY"
    >
      <p
        className="sc-bdvvaa jAWHKi"
        color="#C7C7C7"
      >
        2019年8月
      </p>
    </h6>
    <h6
      className="sc-hKwCoD hglaFY"
    >
      <p
        className="sc-bdvvaa jAWHKi"
        color="#C7C7C7"
      >
        〜
      </p>
    </h6>
    <h6
      className="sc-hKwCoD hglaFY"
    >
      <p
        className="sc-bdvvaa jAWHKi"
        color="#C7C7C7"
      >
        現在
      </p>
    </h6>
  </div>
  <div
    className="sc-jrQzUz dvJrEL"
  >
    <div
      className="sc-jRQAMF hMoLVx"
    >
      <div
        className="sc-gKckTs ibbTIi"
      />
      <div
        className="sc-iCfLBT dRUsRr"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "fontSize": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="sc-kDThTU bXtTaJ"
  >
    <div
      className="sc-iqsfdx MPfKx"
    >
      <h5
        className="sc-dkPtyc liUvOS"
      >
        <p
          className="sc-bdvvaa kxpcMG"
          color="white"
        >
          株式会社xxx
        </p>
      </h5>
      <h6
        className="sc-hKwCoD hglaFY"
      >
        <p
          className="sc-bdvvaa jAWHKi"
          color="#C7C7C7"
        >
          1年間
        </p>
      </h6>
    </div>
    <div
      className="sc-iqsfdx MPfKx"
    >
      <span
        className="sc-crHlIS ejZZvu"
      >
        <div
          className="sc-eCImvq iiYOXJ"
        >
          <p
            className="sc-bdvvaa gceiEI"
            color="#424242"
          >
            営業
          </p>
        </div>
      </span>
      <span
        className="sc-crHlIS ejZZvu"
      >
        <div
          className="sc-eCImvq iiYOXJ"
        >
          <p
            className="sc-bdvvaa gceiEI"
            color="#424242"
          >
            エンジニア
          </p>
        </div>
      </span>
      <span
        className="sc-crHlIS ejZZvu"
      >
        <div
          className="sc-eCImvq iiYOXJ"
        >
          <p
            className="sc-bdvvaa gceiEI"
            color="#424242"
          >
            セールス
          </p>
        </div>
      </span>
      <span
        className="sc-crHlIS ejZZvu"
      >
        <div
          className="sc-eCImvq iiYOXJ"
        >
          <p
            className="sc-bdvvaa gceiEI"
            color="#424242"
          >
            労務
          </p>
        </div>
      </span>
    </div>
    <div
      className="sc-iqsfdx MPfKx"
    >
      <h6
        className="sc-hKwCoD hglaFY"
      >
        <p
          className="sc-bdvvaa kxpcMG"
          color="white"
        >
          helloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworldhelloworld
        </p>
      </h6>
    </div>
  </div>
</div>
`;
